app/src/main/java/com/duckduckgo/app/browser/BrowserViewModel.kt

```diff
@@ -203,7 +206,7 @@ class BrowserViewModel(
 
     fun onOpenShortcut(url: String) {
         launch { tabRepository.selectByUrlOrNewTab(queryUrlConverter.convertQueryToUrl(url)) }
-        if (url == USE_OUR_APP_SHORTCUT_URL) {
+        if (ctaDao.exists(CtaId.USE_OUR_APP) && url == USE_OUR_APP_SHORTCUT_URL) {
```

Comment:
`java.lang.IllegalStateException: Cannot access database on the main thread since it may potentially lock the UI for a long period of time.`

Suggestion: To wrap all the logic inside this method inside that `launch`, and just run it in `Dispatcher.IO`

Reply:
Thanks! I've injected DefaultDispatcherProvider and is all working fine!